app/models/reduction_fetcher.rb

```diff
@@ -1,49 +1,51 @@
-class ReductionFetcher
-  attr_accessor :topic
-  def initialize(filter)
-    @filter = filter
-    @subject_reductions = SubjectReduction.where(filter.except(:user_id))
-    @user_reductions = UserReduction.where(filter.except(:subject_id))
-
-    @topic = :reduce_by_subject
+class ReductionFetcher < FetcherBase
+  attr_reader :loaded
+
+  def initialize(query)
+    super(query)
+    @loaded = false
+
+    @subject_reductions = SubjectReduction.where(query.except(:user_id))
+    @user_reductions = UserReduction.where(query.except(:subject_id))
   end
 
   def load!
+    return self if loaded
```

Comment:
1. no, because the reduction fetcher is shared by all reducers during a given reduction pass
2. it's worth investigating whether adding the ability to load subject reductions and user reductions independently is worth the added complexity in the implementation
3. we absolutely MUST continue pre-fetch these values in order to avoid consistency problems when calculating running reductions.